sdk/identity/src/token_credentials/client_certificate_credentials.rs

```diff
@@ -0,0 +1,231 @@
+use super::{authority_hosts, TokenCredential};
+use azure_core::auth::TokenResponse;
+use chrono::Utc;
+use oauth2::AccessToken;
+use openssl::{
+    error::ErrorStack,
+    hash::{hash, DigestBytes, MessageDigest},
+    pkcs12::Pkcs12,
+    pkey::{PKey, Private},
+    sign::Signer,
+    x509::X509,
+};
+use serde::Deserialize;
+use std::{str, time::Duration};
+
+/// Refresh time to use in seconds
+const DEFAULT_REFRESH_TIME: i64 = 300;
+
+/// Provides options to configure how the Identity library makes authentication
+/// requests to Azure Active Directory.
+#[derive(Clone, Debug, PartialEq)]
+pub struct CertificateCredentialOptions {
+    authority_host: String,
+    send_certificate_chain: bool,
+}
+
+impl Default for CertificateCredentialOptions {
+    fn default() -> Self {
+        Self {
+            authority_host: authority_hosts::AZURE_PUBLIC_CLOUD.to_owned(),
+            send_certificate_chain: false,
+        }
+    }
+}
+
+impl CertificateCredentialOptions {
+    /// Create a new TokenCredentialsOptions. default() may also be used.
+    pub fn new(authority_host: String, send_certificate_chain: bool) -> Self {
```

Comment:
Ideally we take these as `str` but I see this is consistent with our existing credentials. We should fix that later.

Reply:
I can do this in the current PR if you want. not doing it for now